src/main/java/com/microsoft/azure/util/AzureCredentials.java

```diff
@@ -140,7 +140,7 @@ public ServicePrincipal(
             this.clientSecret = Secret.fromString(clientSecret);
             this.oauth2TokenEndpoint = Secret.fromString(oauth2TokenEndpoint);
             this.tenant = Secret.fromString(
-                    ServicePrincipal.getTenantFromTokenEndpoint(oauth2TokenEndpoint));
+                    ServicePrincipal.getTenantFromTokenEndpoint(this.oauth2TokenEndpoint.getPlainText()));
```

Comment:
- parameter `oauth2TokenEndpoint` is a base64-encoded `String` which cannot be handled correctly by 'getTenantFromTokenEndpoint'. 

- `this.oauth2TokenEndpoint` is an instance of `Secret` and its `getPlainText()` returns the decrypted plain text rather than base64 string.